### PR TITLE
Upgrade etcd client to 3.5

### PR DESCRIPTION
**Pull request by brandondr96**
This upgrades the etcd client to 3.5 and fixes the related dependencies which changed. Additionally, since `clientv3` was changed to `client/v3`, I switched the code to reference `v3` for clarity.

### CODE REVIEW

**Review by jswarren4 — CHANGES_REQUESTED** [review]
Requesting changes for now since we're not ready to switch to 3.5 yet.

**Comment by kramvan1** [comment]
Need to revert these 3.5 backports now https://github.com/IBM-Cloud/go-etcd-rules/pull/122

**Review by kramvan1 — CHANGES_REQUESTED** [review]
Need to revert https://github.com/IBM-Cloud/go-etcd-rules/pull/122

**Review by jswarren4 — DISMISSED** [review]
Approving so my change request doesn't block merge.

File: Makefile [hunk 1]
```diff
@@ -30,7 +30,7 @@ test: int-setup
 
 .PHONY: int-setup
 int-setup: int-teardown
-	docker run -d -p 2379:2379 --name etcd quay.io/coreos/etcd:v3.4.18 \
+	docker run -d -p 2379:2379 --name etcd quay.io/coreos/etcd:v3.5.1 \
```
Comment (kramvan1) [inline Makefile:33]: Can make this 3.5.2 now

File: rules/engine_test.go [hunk 2]
```diff
@@ -34,6 +34,17 @@ func TestV3EngineConstructor(t *testing.T) {
 	err := eng.AddPolling("/polling", rule, 30, v3DummyCallback)
 	assert.NoError(t, err)
 	eng.Run()
+	eng.Stop()
+	stopped := false
+	for i := 0; i < 60; i++ {
```
Comment (kramvan1) [inline rules/engine_test.go:39]: Might be nice to push this into a helper function.

**Review by kramvan1 — DISMISSED** [review]

File: go.mod [hunk 3]
```diff
@@ -15,26 +16,22 @@ require (
...
-	github.com/golang/protobuf v1.4.3 // indirect
-	github.com/google/uuid v1.3.0 // indirect
+	github.com/gogo/protobuf v1.3.2 // indirect
+	github.com/golang/protobuf v1.5.2 // indirect
 	github.com/matttproud/golang_protobuf_extensions v1.0.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/prometheus/client_model v0.2.0 // indirect
 	github.com/prometheus/common v0.26.0 // indirect
 	github.com/prometheus/procfs v0.8.0 // indirect
+	go.etcd.io/etcd/client/pkg/v3 v3.5.7 // indirect
```
Comment (sonaljain0) [inline go.mod:28]: Shouldn't this be 3.5.6?

File: go.sum [hunk 4]
```diff
@@ -153,86 +129,89 @@ github.com/prometheus/procfs v0.1.3/go.mod h1:lV6e/gmhEcM9IjHGsFOCxxuZ+z1YqCvr4O
...
+go.etcd.io/etcd/api/v3 v3.5.6/go.mod h1:KFtNaxGDw4Yx/BA4iPPwevUTAuqcsPxzyX8PHydchN8=
+go.etcd.io/etcd/api/v3 v3.5.7 h1:sbcmosSVesNrWOJ58ZQFitHMdncusIifYcrBfwrlJSY=
+go.etcd.io/etcd/api/v3 v3.5.7/go.mod h1:9qew1gCdDDLu+VwmeG+iFpL+QlpHTo7iubavdVDgCAA=
+go.etcd.io/etcd/client/pkg/v3 v3.5.6/go.mod h1:ggrwbk069qxpKPq8/FKkQ3Xq9y39kbFR4LnKszpRXeQ=
+go.etcd.io/etcd/client/pkg/v3 v3.5.7 h1:y3kf5Gbp4e4q7egZdn5T7W9TSHUvkClN6u+Rq9mEOmg=
+go.etcd.io/etcd/client/pkg/v3 v3.5.7/go.mod h1:o0Abi1MK86iad3YrWhgUsbGx1pmTS+hrORWc2CamuhY=
+go.etcd.io/etcd/client/v3 v3.5.6 h1:coLs69PWCXE9G4FKquzNaSHrRyMCAXwF+IX1tAPVO8E=
+go.etcd.io/etcd/client/v3 v3.5.6/go.mod h1:f6GRinRMCsFVv9Ht42EyY7nfsVGwrNO0WEoS2pRKzQk=
+go.etcd.io/etcd/client/v3 v3.5.7 h1:u/OhpiuCgYY8awOHlhIhmGIGpxfBU/GZBUP3m/3/Iz4=
+go.etcd.io/etcd/client/v3 v3.5.7/go.mod h1:sOWmj9DZUMyAngS7QQwCyAXXAL6WhgTOPLNS/NabQgw=
```
Comment (sonaljain0) [inline go.sum:160]: Lots of 3.5.7 here

**Review by sonaljain0 — CHANGES_REQUESTED** [review]
Added comments

**Review by sonaljain0 — APPROVED** [review]

**Pull request merged**